main/inc/lib/main_api.lib.php

```diff
@@ -7498,6 +7511,63 @@ function api_register_campus($listCampus = true) {
 }
 
 /**
+ * Check whether the user type should be exclude.
+ * Such as invited or anonymous users
+ * @param boolean $checkDB Optional. Whether check the user status
+ * @param int $userId Options. The user id
+ * @return boolean
+ */
+function apiIsExcludedUserType($checkDB = false, $userId = 0)
+{
+    if ($checkDB) {
+        $userId = empty($userId) ? api_get_user_id() : intval($userId);
+
+        if ($userId == 0) {
+            return true;
+        }
+
+        $userInfo = api_get_user_info($userId);
+
+        switch ($userInfo['status']) {
+            case ROLE_INVITED:
+                //no break;
+            case ANONYMOUS:
+                return true;
+            default:
+                return false;
+        }
+    }
+
+    $isInvited = apiIsInvitedUser();
+    $isAnonymous = api_is_anonymous();
+
+    if ($isInvited || $isAnonymous) {
+        return true;
+    }
+
+    return false;
+}
+
+/**
+ * Get the exclude user types
+ * @param string $format Optional. The result type. Array or string
+ * @return array
+ */
+function apiGetExcludedUserTypes($format = 'array')
```

Comment:
This function name is very tricky. 
- Those are not user types, those are user status.
- Get excluded user status, excluded from what? 
- The name should changed for something more descriptive.